src/test/java/org/opensearch/ad/indices/CustomIndexTests.java

```diff
@@ -52,31 +54,28 @@ public void setUp() throws Exception {
 
         client = mock(Client.class);
 
-        clusterService = mock(ClusterService.class);
+        clusterService = mock(SDKClusterService.class);
 
         clusterName = new ClusterName("test");
 
         customIndexName = "opensearch-ad-plugin-result-a";
-        // clusterState = ClusterState.builder(clusterName).metadata(Metadata.builder().build()).build();
+        clusterState = ClusterState.builder(clusterName).metadata(Metadata.builder().build()).build();
         when(clusterService.state()).thenReturn(clusterState);
 
         Settings settings = Settings.EMPTY;
-        ClusterSettings clusterSettings = new ClusterSettings(
-            settings,
-            Collections
-                .unmodifiableSet(
-                    new HashSet<>(
-                        Arrays
-                            .asList(
-                                AnomalyDetectorSettings.AD_RESULT_HISTORY_MAX_DOCS_PER_SHARD,
-                                AnomalyDetectorSettings.AD_RESULT_HISTORY_ROLLOVER_PERIOD,
-                                AnomalyDetectorSettings.AD_RESULT_HISTORY_RETENTION_PERIOD,
-                                AnomalyDetectorSettings.MAX_PRIMARY_SHARDS
-                            )
-                    )
-                )
-        );
-
+        List<Setting<?>> settingsList = List
+            .of(
+                AnomalyDetectorSettings.AD_RESULT_HISTORY_MAX_DOCS_PER_SHARD,
+                AnomalyDetectorSettings.AD_RESULT_HISTORY_ROLLOVER_PERIOD,
+                AnomalyDetectorSettings.AD_RESULT_HISTORY_RETENTION_PERIOD,
+                AnomalyDetectorSettings.MAX_PRIMARY_SHARDS
+            );
+        ExtensionsRunner mockRunner = mock(ExtensionsRunner.class);
+        Extension mockExtension = mock(Extension.class);
+        when(mockRunner.getEnvironmentSettings()).thenReturn(settings);
+        when(mockRunner.getExtension()).thenReturn(mockExtension);
+        when(mockExtension.getSettings()).thenReturn(settingsList);
```

Comment:
Thinking it out loud. It's definitely good to have more tests and checks but since we are focusing on migration. Should we add other checks except the one below?

Reply:
Right now I'm trying to restore exactly the same test functionality that already exists.  Not sure what needs to be added that you wouldn't want to also add to the main project.

Reply:
Let me be more specific here. I meant to ask do we need to test these because to reduce the diff in terms of migration these tests are extra addition :

```
 List<Setting<?>> settingsList = List.of(AnomalyDetectorSettings.MAX_ENTITIES_FOR_PREVIEW, AnomalyDetectorSettings.PAGE_SIZE);
        ExtensionsRunner mockRunner = mock(ExtensionsRunner.class);
        Extension mockExtension = mock(Extension.class);
        when(mockRunner.getEnvironmentSettings()).thenReturn(settings);
        when(mockRunner.getExtension()).thenReturn(mockExtension);
        when(mockExtension.getSettings()).thenReturn(settingsList);

Reply:
I think I understand.  Here's the dilemma:
 - I implemented the `SDKClusterSettings` as an inner class, so it would have access to the `SDKClusterService` instance field of ExtensionsRunner.
 - The reason for this is that the Cluster Settings object needs methods involving transport (settings update consumers, etc.) that require the ExtensionsRunner or other objects from it.
 - This means we can't just instantiate it like a regular class: that would require it to be a static (nested) class.  But instantiating it like above would not allow it to have access to the Extensions Runner.
 - There are workarounds to all of the above, but generally speaking I think minimizing the diff in the src/main/java filepath is worth it for a few boilerplate copy/paste mocking workarounds in a small number of test classes (we could even create a separate single mock somewhere to minimize repetition if it gets out of hand).

I welcome you to propose a better way to do this.  Believe me I tried many different options and settled on this one, but TMTOWTDI.
 - So matching the existing constructor taking arguments that do not include
 - to match the existing test (and reduce the diff

Reply:
I liked the idea of creating a separate single mock if migration changes are creating more diff. We can think about this later.